src/components/CvTimePicker/CvTimePicker.vue

```diff
@@ -0,0 +1,217 @@
+  },
+  /**
+   * Label for the AM/PM `<cv-select>`
+   */
+  ampmSelectLabel: { type: String, default: 'Select AM/PM' },
+  /**
+   * Specify whether the `<input>` should be disabled
+   */
+  disabled: { type: Boolean, default: false },
+  /**
+   * Provide the text that is displayed when the control is in an invalid state
+   */
+  invalidMessage: { type: String, default: undefined },
+  /**
+   * Label for the time `<input>`
+   */
+  label: { type: String, default: 'Select a time' },
+  /**
+   * Specify the regular expression working as the [pattern](https://www.w3schools.com/tags/att_input_pattern.asp) of the time string in `<input>`.
+   */
+  pattern: { type: String, default: '([01][0-9]:[0-6][0-9])' },
+  /**
+   * Specify the placeholder attribute for the <input>
+   */
+  placeholder: { type: String, default: 'hh:mm' },
+  /**
+   * Specify the initial value of the `<input>`
+   */
+  time: { type: String, default: undefined },
+  /**
+   * Specify the initial value of the `<cv-select>` for the timezone
+   */
+  timezone: { type: String, default: undefined },
+  /**
+   * List of timezones to display
+   * @type {Array<{value: string, label:string}>}
+   */
+  timezones: {
+    type: Array,
+    default: () => [],
+  },
+  /**
+   * Specify the label of the `<cv-select>` for the timezone. Label is hidden but text will be read by a screen reader when visiting this control.
+   */
+  timezonesSelectLabel: { type: String, default: 'Select time zone' },
+  /**
+   * Specify the size of the Time Picker.
+   */
+  fieldSize: {
+    type: String,
+    default: FIELD_MEDIUM,
+    validator: value => TIME_PICKER_SIZES.includes(value),
+  },
+  ...propsCvId,
+  ...propsTheme,
+});
+
+const uid = useCvId(props, true);
+const isLight = useIsLight(props);
+
+const isInvalid = ref(false);
+const slots = useSlots();
+function checkSlots() {
+  // NOTE: slots is not reactive so needs to be managed on updated
+  isInvalid.value = !!(
+    slots['invalid-message'] ||
+    (props.invalidMessage && props.invalidMessage.length)
+  );
+}
+onMounted(checkSlots);
+onUpdated(checkSlots);
+
+const emit = defineEmits([
+  'change',
+  'update:ampm',
+  'update:timezone',
+  'update:time',
+]);
+
+const validTimezone = computed(() => {
+  // Validate timezone setting
+  let result = props.timezone;
+  if (props.timezones && props.timezones.length) {
+    if (!props.timezones.find(item => item.value === props.timezone)) {
+      if (props.timezone !== undefined)
+        console.error(
+          `CvTimePicker: invalid value '${props.timezone}' supplied for prop timezone. Default applied.`
+        );
+      // set to first valid value
+      result = props.timezones[0].value;
+      emit('update:timezone', result);
+    }
+  }
+  return result;
+});
+const isDisabled = computed(() => (props.disabled ? true : undefined));
+
+function onInput(e) {
+  emit('change', { type: 'time', value: e?.target?.value });
```

Comment:
suggestion for later: changes types might be a candidate to enum (or to be precise js object that tries to pretend to be enum :) ) that will help users to detect change type in a consistent way

eg: 
const TimePickerChangeTypes = {
time: 'time',
ampm: 'ampm',
...
}